subcommands/keys/rotate_root.go

```diff
@@ -48,6 +50,27 @@ func doRotateRoot(cmd *cobra.Command, args []string) {
 	subcommands.DieNotNil(err)
 
 	if initialRotation {
+		if _, err := os.Stat(credsFile); err == nil {
+			subcommands.DieNotNil(errors.New("Destination file exists. Please make sure you aren't accidentally overwriting another factory's keys"))
+		}
+
+		var checkPath func(string)
+
+		checkPath = func(path string) {
+			_, err := os.Stat(path)
+			if os.IsNotExist(err) {
+				subcommands.DieNotNil(fmt.Errorf("Destination path does not exist: %v", path))
+				return
+			} else if os.IsPermission(err) {
+				subcommands.DieNotNil(fmt.Errorf("Destination path is not writable: %v", path))
+				return
+			}
+
+			checkPath(filepath.Dir(path))
+		}
+
+		checkPath(filepath.Dir(credsFile))
```

Comment:
this is cluttering up the main function and should be broken out, but I don't think it really adds much value. If we skip the check you already get: "ERROR: /tmp/blah/foo: no such file or directory" that seems sufficient, no?

Reply:
Reverted :+1:

Reply:
I just ran the command on a path that contains non-existent directories, but it will fail after already getting the files for key rotation. Is that an issue?

Reply:
nope. it can re-run and get them again. https://git.foundries.io/development/cloudplatforms/infrastructure/ota-lite/-/blob/master/ota_lite/api/factory.py#L446-463

Reply:
as a general note about stuff kind of like this. you'll see code all the time like:
```
if file.exists():
   content = open(file).read()
```
it isn't foolproof. Its mostly fine on single-threaded single-user systems, but its easier to just rely on the OS with something like:
```
try:
   content = open(file).read()
except FileNotFoundError
```

Reply:
Gotcha. Then I will merge it as is. :smile:

Reply:
Sorry for confusion from my side
I really only cared about loosing keys, not the error message.
A fix as is seems to close all the remaining gaps, so it's fine... I gave my LGTM yesterday.